### PR TITLE
[MIG][10.0] Porting partner_street_number to V10

### PR DESCRIPTION
**Pull request by astirpe**
Based on the job https://github.com/OCA/partner-contact/pull/207 made by @DarkoNikolovski.

### CODE REVIEW

File: partner_street_number/__init__.py
Comment (hbrunn) [inline partner_street_number/__init__.py]: add `.with_context(active_test=False)` to get all partners here.
Search can be formatted more beautifully as

``` python
myobject.search([
    (prop1),
    (prop2),
])
```

making a change in a proposition a one line change

File: partner_street_number/__init__.py
Comment (hbrunn) [inline partner_street_number/__init__.py]: your environment is already created for root, so no need to sudo here a second time

File: partner_street_number/static/src/css/field.css
Comment (hbrunn) [inline partner_street_number/static/src/css/field.css]: is is possible to avoid the `!important`s by adding a longer selector? The important qualifier tends to mess up css in the sense that it overrides everything

**Review by hbrunn — CHANGES_REQUESTED** [review]

File: partner_street_number/__manifest__.py
Comment (pedrobaeza) [inline partner_street_number/__manifest__.py]: 10.0.1.0.0

File: partner_street_number/i18n/en.po
Comment (pedrobaeza) [inline partner_street_number/i18n/en.po]: Remove this file

File: partner_street_number/i18n/partner_street_number.pot
Comment (pedrobaeza) [inline partner_street_number/i18n/partner_street_number.pot]: Remove this file

File: partner_street_number/static/src/css/field.css
Comment (pedrobaeza) [inline partner_street_number/static/src/css/field.css]: Maybe you can make a less?

**Review by pedrobaeza — APPROVED** [review]
Well, the less is the same of the CSS. It doesn't make use of any of the available variables that can be changed in pack, but for now I'm not going to push further.

Please squash on merging.

**Review by rafaelbn — APPROVED** [review]
Tested in runbot

**Comment by pedrobaeza** [comment]
You should now that Odoo is working on a similar approach with the recent announce of including Mexican localization in v10 EE: https://github.com/odoo/odoo/pull/13835.

File: partner_street_number/models/res_partner.py
Comment (OpenCode) [inline partner_street_number/models/res_partner.py]: Can you use inline if:

`partner.street.strip() if partner.street else False`

?

File: partner_street_number/views/res_partner.xml
Comment (OpenCode) [inline partner_street_number/views/res_partner.xml]: <openerp><data> can be replaced with <odoo>

**Comment by rafaelbn** [comment]
Hi @hbrunn , here is missing you final review, thanks!

File: partner_street_number/__manifest__.py
Comment (dreispt) [reply partner_street_number/__manifest__.py]: why?

**Comment by pedrobaeza** [comment]
We should wait for merging this module until https://github.com/odoo/odoo/pull/13835 is resolved, because this overlaps the same functionality in core.

**Review by hbrunn — APPROVED** [review]

**Comment by hbrunn** [comment]
I'd propose to merge as it is, and adapt to core changes when core is actually changed

**Comment by pedrobaeza** [comment]
@hbrunn I prefer not, or will have to deal later with migration scripts and maybe to have a dummy module all version around.

**Comment by astirpe** [comment]
@pedrobaeza what about all the modules that depend on this?

**Comment by hbrunn** [comment]
I don't see a problem with needing migration scripts, and replacing all address-related modules from this repo by dummies once `base_address_structured` landed seems the right thing to do to me. We have no clue how long it will take, so if somebody moves forward before, why block it?

**Comment by pedrobaeza** [comment]
Please just wait one week to see if there are moves upstream. Then we decide.

**Comment by hbrunn** [comment]
one week, deal. Revisiting October 28th ;-)

**Comment by astirpe** [comment]
and...?

**Comment by pedrobaeza** [comment]
hehe, OK, let's go with this one, as there's no answer from Odoo so far.

**Comment by pedrobaeza** [comment]
Anyway, maybe you can consider the same things as I say in the other PR: https://github.com/odoo/odoo/pull/13835#discussion_r84131784

**Comment by hbrunn** [comment]
I say: Merge now, and add @pedrobaeza's suggestions which sound perfectly valid to me as occasion allows

File: partner_street_number/hooks.py
Comment (pedrobaeza) [inline partner_street_number/hooks.py]: Please short header.

**Comment by astirpe** [comment]
Indeed the comments of @pedrobaeza are interesting. We could get inspiration from https://github.com/openvenues/libpostal and its Python bindings to make a generic solution. Thank you!

I will create a new PR for that. Or maybe someone else would like to work at it?

**Comment by hbrunn** [comment]
@pedrobaeza can we merge this?

**Comment by pedrobaeza** [comment]
OK, let's go!

**Pull request merged**
